src/groups/bmq/bmqp/bmqp_protocol.h

```diff
@@ -4353,9 +4352,9 @@ inline AckHeader& AckHeader::setHeaderWords(int value)
     BSLS_ASSERT_SAFE(value >= 0 &&
                      value <= (1 << k_HEADER_WORDS_NUM_BITS) - 1);
 
-    d_headerWordsAndPerMsgWords =
```

Comment:
Only thing is that k_PER_MSG_WORDS_MSK and k_HEADER_WORDS_START_IDX is also used elsewhere in the file (12 & 16 times respectively), so may not make sense to change to change the source. However, I think it may still be possible to just use one cast for cleanness which I'll double check with the CI.

Reply:
@pniedzielski confirmingthat CI checks passed after removing the nested cast. We're now down to 1,625 and 2,513 in docker & ubuntu builds respectively (from tens of thousands originally)!